### PR TITLE
[node-provider] Adds initial scaffolding for Node Provider package

### PR DESCRIPTION
**Pull request by joelalejandro**
### Description

This PR allows work to begin on the NodeProvider package, adding basic dependencies and directory structure.

### CODE REVIEW

File: packages/node-provider/package.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "@counterfactual/node-provider",
+  "version": "0.0.1",
+  "main": "dist/index.js",
+  "types": "dist/src/index.d.ts",
+  "module": "dist/index.es.js",
+  "iife": "dist/index-iife.js",
```
Comment (snario) [inline packages/node-provider/package.json]: Do we need all of these outputs? Specifically `.es` and `.iife`?

Comment (joelalejandro) [reply packages/node-provider/package.json]: @snario Not necessarily. I just copied from what we have in `packages/machine`:

https://github.com/counterfactual/monorepo/blob/50b6df871235a9f687a5dd765328c8e4c66a6507/packages/machine/package.json#L1-L7

File: packages/node-provider/tslint.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": ["../../tslint.json"]
+  "extends": ["../../tslint.json"],
+  "rules": {
+    "import-name": false
```
[the remark this answers is not in the document]
Comment (joelalejandro) [reply packages/node-provider/tslint.json]: Class names are being incorrectly marked as "error". If I create a file "node-provider.ts", it expects the class name to be called "nodeProvider". All class names should be capitalized.

File: packages/node-provider/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import typescript from "rollup-plugin-typescript2";
```
[the remark this answers is not in the document]
Comment (joelalejandro) [reply packages/node-provider/rollup.config.js]: Should Playground add this package as a dependency and consume it as TypeScript directly?

File: packages/node-provider/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": ".",
+    "sourceRoot": ".",
```
[the remark this answers is not in the document]
Comment (joelalejandro) [reply packages/node-provider/tsconfig.json]: Probably not. I'll remove it.

File: packages/node-provider/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    "declarationDir": "dist/",
```
[the remark this answers is not in the document]
Comment (joelalejandro) [reply packages/node-provider/tsconfig.json]: Copied. Probably no need to have it.

File: packages/node-provider/rollup.config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (snario) [reply packages/node-provider/rollup.config.js]: Well, at the moment, the _only_ consumer of this thing is the Playground. So, probably this is fine for now until we have a second consumer with a more complicated use case. But in general I'm in favour of keeping things as simple as possible until there is a need to add complexity.

File: packages/node-provider/tslint.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (snario) [reply packages/node-provider/tslint.json]: Apparently it changes depending on how you use it? https://github.com/Microsoft/tslint-microsoft-contrib/issues/387#issuecomment-402985955

[the remark this answers is not in the document]
Comment (joelalejandro) [reply packages/node-provider/tslint.json]: Well, that's too weird of a behaviour. Basically the linter will recorgnize it as a class of it is instantiated somewhere. Sometimes that won't happen. I don't feel too comfortable with this rule actually. It's more of a hassle than a convenience.

[the remark this answers is not in the document]
Comment (snario) [reply packages/node-provider/tslint.json]: Yeah sure I'm fine with that. We should probably make it a rule at the root level tslint then

[the remark this answers is not in the document]
Comment (joelalejandro) [reply packages/node-provider/tslint.json]: Great, will do!

**Pull request merged**